commit eb72c1faedc39c68fb1246ea8a97d1f96831756c
Author: Willy Tarreau <w@1wt.eu>
Date:   Mon Oct 15 11:12:15 2018 +0200

    BUILD: stick-table: make sure not to fail on task_new() during initialization
    
    Gcc reports a potential null-deref error in the stick-table init code.
    While not critical there, it's trivial to fix. This check has been
    missing since 1.4 so this fix can be backported to all supported versions.
    
    (cherry picked from commit 848522f05df9e60eea9274e11f1e9fcd19594a5c)
    Signed-off-by: Willy Tarreau <w@1wt.eu>

diff --git a/src/stick_table.c b/src/stick_table.c
index 5a2f1295..653a1ffb 100644
--- a/src/stick_table.c
+++ b/src/stick_table.c
@@ -602,6 +602,8 @@ int stktable_init(struct stktable *t)
 		t->exp_next = TICK_ETERNITY;
 		if ( t->expire ) {
 			t->exp_task = task_new(MAX_THREADS_MASK);
+			if (!t->exp_task)
+				return 0;
 			t->exp_task->process = process_table_expire;
 			t->exp_task->context = (void *)t;
 		}
